library/src/hcc_detail/hipsolver_compat.cpp

```diff
@@ -0,0 +1,139 @@
+/* ************************************************************************
+ * Copyright 2020-2021 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "hipsolver.h"
+
+extern "C" {
+
+// gesvd
+hipsolverStatus_t hipsolverDnSgesvd_bufferSize(hipsolverDnHandle_t handle, int m, int n, int* lwork)
+{
+    return hipsolverSgesvd_bufferSize(handle, 'N', 'N', 0, 0, lwork);
```

Comment:
Similarly to getrf, I think this helper should return a non-zero value when 'm' and 'n' are not zero. Here the things are more complicated, though, as we don't know the values of `jobu` and `jobv`. I see two options:
a. Get the workspace size for all the 15 possible permutations of `jobu` and `jobv`, and return the maximum.
b. Modify `gesvd_getMemorySize` in rocsolver to always return the maximum regardless of `jobu` and `jobv` values.

I don't think that the difference in size for the different values of `jobu` and `jobv` is substantial if any. So, we could do option b and, eventually, as the values of `jobu` and `jobv` will become useless, we could then remove them altogether from the APIs. But this sound like a longer path, so for now, I would suggest we follow option a. What do you think?